source/slang/slang-check-decl.cpp

```diff
@@ -7833,6 +8045,73 @@ namespace Slang
         return as<SeqStmt>(stmt->body);
     }
 
+    static bool _doesCtorExpectInitializerListUsage(ConstructorDecl* ctor)
+    {
+        return ctor->containsOption(ConstructorTags::Synthesized)
+            && ctor->getParameters().getCount() != 0
+            && !allParamHaveInitExpr(ctor);
+    }
+
+    template<typename T>
+    bool _containsTargetType(ASTBuilder* m_astBuilder, NodeBase* target, HashSet<NodeBase*>& visitedNodes)
```

Comment:
this is very heavy weight recursion, why do we need this?

Reply:
We need this or a cached version of some sort (I agree this is quite an expensive function) to detect if we need our ctor to be tagged with "CudaHost" due to CudaHost only type being found.

Reply:
https://github.com/shader-slang/slang/pull/4854/files/c0e59e84f30557389d8dd895198c068c219fb6f7#r1747817868

A legalization pass for 1 target is a better idea than this expensive recursive routine.